File: components/common/src/androidMain/kotlin/com/mirego/pilot/components/ui/type/PilotKeyboardReturnKeyType.kt
package com.mirego.pilot.components.ui.type

import androidx.compose.ui.text.input.ImeAction
import com.mirego.pilot.components.InternalPilotComponentsApi
import com.mirego.pilot.components.type.PilotKeyboardReturnKeyType

@InternalPilotComponentsApi
public val PilotKeyboardReturnKeyType.composeValue: ImeAction
    get() = when (this) {
        PilotKeyboardReturnKeyType.Default -> ImeAction.Default
        PilotKeyboardReturnKeyType.Done -> ImeAction.Done
        PilotKeyboardReturnKeyType.Go -> ImeAction.Go
        PilotKeyboardReturnKeyType.Next -> ImeAction.Next
        PilotKeyboardReturnKeyType.Search -> ImeAction.Search
        PilotKeyboardReturnKeyType.Send -> ImeAction.Send
    }

File: components/ios/base/Types/PilotKeyboardReturnKeyType.swift
import Shared
import SwiftUI

extension PilotKeyboardReturnKeyType {
    @available(iOS 15.0, *)
    public var submitLabel: SubmitLabel {
        switch self {
        case .default:
            return .done
        case .done:
            return .done
        case .go:
            return .go
        case .next:
            return .next
        case .search:
            return .search
        case .send:
            return .send
        }
    }
}
